client_integration_test.go

```diff
@@ -1489,6 +1489,49 @@ func TestClientReadFrom(t *testing.T) {
 	}
 }
 
+func TestClientReadFromSectionReader(t *testing.T) {
+	const (
+		packetSize = 1024
+		bufsize    = 4 * packetSize
+		skip       = 100
+		filesize   = bufsize - skip
+	)
+
+	sftp, cmd := testClient(t, READWRITE, NODELAY,
+		MaxPacketChecked(packetSize),
+		UseConcurrentWrites(true))
+	defer cmd.Wait()
+	defer sftp.Close()
+
+	d, err := ioutil.TempDir("", "sftptest-readfrom-sectionreader")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(d)
+
+	buf := bytes.NewReader(make([]byte, bufsize))
+	r := io.NewSectionReader(buf, 0, bufsize)
+
+	// Skip some bytes so that r.Size is no longer accurate.
+	// ReadFrom uses it to determine concurrency,
+	// but should not rely on it to determine how many bytes to send.
+	r.Seek(skip, io.SeekStart)
+
+	f := path.Join(d, "writeTest")
+	w, err := sftp.Create(f)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer w.Close()
+
+	n, err := w.ReadFrom(r)
+	assert.EqualValues(t, bufsize-skip, n)
```

Comment:
`bufsize-skip` is the same value as `filesize`

Ooo… `EqualValues` is a nice function over `Equal` which often requires you to cast your constants to specific values.

Reply:
Changed to filesize.